### ISSUE TITLE
MatcherUtils uses reflection

### ISSUE THREAD

**Issue by quat1024** [quat1024/incorporeal-2-forge#4]
Generally serializing and deserializing ICorporeaRequestMatchers is sorta messy and i should talk to botania people about this

**Comment by Hubry** [comment]
Is it fine if we just provide these three:
```
CorporeaHelper#getMatcherId(ICorporeaRequestMatcher matcher)
CorporeaHelper#getMatcherId(Class<? extends ICorporeaRequestMatcher> clazz)
CorporeaHelper#readMatcher(ResourceLocation id, CompoundNBT cmp)
```
~~Also your reflection is ugly as heck, you could just stick these maps into a static field and not repeat the reflective access more than once. Also static accessors work just fine. :wacko:~~

**Comment by quat1024** [comment]
fixed in 99c6607013e8c284dbec024b918c3195eba0b232 to make it not use reflection & talked on discord and `i guess it's good enough for now` maybe look into a nicer api in 1.17
